starcounter-upload.html

```diff
@@ -344,7 +353,14 @@
                     this.fire("statechange", task);
                     this.processTask(task);
                 }
-            }
+            },
+            checkFile: function (file) {
+                if (file.size > this.maxFileSize && this.maxFileSize > -1) {
+                    alert("File " + file.name + " is too big. Maximum file size is: " + this.maxFileSize + "kb.");
```

Comment:
The [`file.size`](https://developer.mozilla.org/en-US/docs/Web/API/Blob/size) property is specified in bytes, not kilobytes.

Reply:
Using `alert` is evil. Modern browsers usually block alerts, and users will get frustrated without any feedback on why the file is not uploading. I would rather fire the `statechange` event with `fail to upload` state.